src/swagger/specGenerator.ts

```diff
@@ -35,30 +36,111 @@ export class SpecGenerator {
     return methodName.charAt(0).toUpperCase() + methodName.substr(1);
   }
 
-  protected getSwaggerType(type: Tsoa.Type): Swagger.Schema {
-    const swaggerType = this.getSwaggerTypeForPrimitiveType(type);
-    if (swaggerType) {
-      return swaggerType;
+  public throwIfNotDataFormat(strToTest: string): Swagger.DataFormat {
+    const guiltyUntilInnocent = strToTest as Swagger.DataFormat;
+    if (
```

Comment:
The cool thing about this change is that the compiler (via `assertNever`) will inform us on where changes are necessary if someone were to add a new type to the `Swagger.DataFormat` union.